### ISSUE TITLE
Aid Treeshaking by importing only what is needed

### ISSUE THREAD

**Issue by shaula** [openstreetmap/iD#3948]
There are two big dependencies within iD, one is lodash, the other is d3. By using:
`import * as d3 from 'd3'` everything is imported. Within a modern setup using Rollup or Webpack2 this hinders Treeshaking to perform well. If only e.g. the dispatch function is used from d3 within an iD module, then its just fine to import:
`import dispatch from 'd3-dispatch';`
If this is done consequently, the build system is able to get rid of unused modules easily, saving tons of KBs. While d3 is already split into various modules, for lodash we would have to switch to https://www.npmjs.com/package/lodash-es
Any thoughts or blockers on this issue?

**Comment by bhousel** [comment]
Yes, when we did #3257 and #3344, I had the idea that we would eventually replace those `import *` with better imports that just target the modules that are really needed.  Thanks for opening this issue!

There aren't any real blockers, but it would probably be good to wait until we release 2.2 and then clear out some of the PR backlog before changing lots of files.

**Comment by bhousel** [comment]
I did some work on this which lives here:  https://github.com/openstreetmap/iD/pull/4372#issuecomment-332292559

Leaving this issue open because the bundle didn't actually shrink at all and there are other blockers now - but maybe we are closer!
